gateway/metrics/add_metrics.go

```diff
@@ -34,15 +34,13 @@ func AddMetricsHandler(handler http.HandlerFunc, prometheusQuery PrometheusQuery
 			log.Printf("List functions responded with code %d, body: %s",
 				recorder.Code,
 				string(upstreamBody))
-
-			http.Error(w, "Metrics handler: unexpected status code from provider listing functions", http.StatusInternalServerError)
+			http.Error(w, string(upstreamBody), recorder.Code)
```

Comment:
I think I am OK with this change, let's hope nobody is looking for that specific message because I think it's been there a long time.